### ISSUE TITLE
Boundary not implemented in Sprite / PhysicsEngineSimple

### ISSUE THREAD

**Issue by TheFonus** [pvcraven/arcade#281]
## Bug Report
Boundary not implemented in Sprite / PhysicsEngineSimple

### Actual behavior:
**For Sprite**:
Initiated in __init__ as:
self.boundary_left = None
and stays as None
 (same for right, top, bottom)

**For PhysicsEngineSimple:**
Uses player_sprite.left instead of player_sprite.boundary_left for collision detection.
(same for right, top, bottom)

### Expected behavior:
Boundary exposed as property and used by PhysicsEngineSimple.

### Steps to reproduce/example code:
sprite.center_x = 10
print(sprite.left)
print(sprite.boundary_left)

## Enhancement request:
Basically I want to be able to have the collision area to differ from the Sprite area.

**Comment by pvcraven** [comment]
`boundary_left` is just for keeping left/right boundaries for sprite that moves back and forth.

The 'points' allows you select a different collision area. See the 'ramps' example.
